src/file_pipe_log/pipe_builder.rs

```diff
@@ -528,13 +528,14 @@ impl<F: FileSystem> DualPipesBuilder<F> {
 
     fn initialize_files(&mut self) -> Result<()> {
         let target_file_size = self.cfg.target_file_size.0 as usize;
-        let mut target = std::cmp::min(
-            self.cfg.prefill_capacity(),
-            self.cfg
-                .recycle_capacity()
-                .saturating_sub(self.append_files.len()),
-        );
-        let to_create = target.saturating_sub(self.recycled_files.len());
+        let prefill_target =
+            std::cmp::min(self.cfg.prefill_capacity(), self.cfg.recycle_capacity())
```

Comment:
This is not the same as old code.

Reply:
Because the old code is not fit anymore after the recycle threshold is enlarged. From @LykxSassinator, the old logic is that `prefill_capacity` is the number of recycle files that should be pre-created even if there are already some append files; but the new logic changes the semantic to let `prefill_capacity` the total files number (recycle + append), I think this is more intuitive.  
In the general case, we expect the total files size is around the size of purge threshold but not the recycle threshold ( 1.5 * purge_threshold), so it's better not to pre-create too many redundant files.

Reply:
> we expect the total files size is around the size of purge threshold but not the recycle threshold ( 1.5 * purge_threshold)

I don't think so. We need to prepare the same amount of files as recycle-threshold. The condition where a large recycle-threshold is needed can also occur when raft-engine has just started.

Reply:
I think there is a bug in the old code that it only count the recycle file to create prefill files count. While with the default config there is no problem because it is equal to the recycle threshold. But if the prefill threshold is smaller than recycle threshold, I can cause raft-engine to create more files after restarting which is unexpacted. One of the log recycle test case failed after I exlarged the recycle threshold.